CMakeLists.txt

```diff
@@ -460,109 +470,114 @@ if(MSVC)
-                (void)c;
+                unsigned crc = 0;
+                char c = 'c';
+            #if defined(_MSC_VER)
+                crc = _mm_crc32_u32(crc, c);
+            #else
+                crc = __builtin_ia32_crc32qi(crc, c);
+            #endif
+                (void)crc;
                 return 0;
             }"
-            HAVE_PCLMULQDQ_INTRIN
+            HAVE_SSE42CRC_INTRIN
         )
-    else()
-        set(HAVE_PCLMULQDQ_INTRIN NO)
+        set(CMAKE_REQUIRED_FLAGS)
     endif()
-    set(CMAKE_REQUIRED_FLAGS)
-
-    # Check whether compiler supports AVX2 intrinics
-    if(WITH_NATIVE_INSTRUCTIONS)
-        set(CMAKE_REQUIRED_FLAGS "${NATIVEFLAG}")
-    else()
-        set(CMAKE_REQUIRED_FLAGS "${AVX2FLAG}")
+    if(WITH_PCLMULQDQ)
+        # Check whether compiler supports PCLMULQDQ intrinics
+        if(WITH_NATIVE_INSTRUCTIONS)
+            set(CMAKE_REQUIRED_FLAGS "${NATIVEFLAG}")
+        else()
+            set(CMAKE_REQUIRED_FLAGS "${PCLMULFLAG}")
+        endif()
+        if(NOT (APPLE AND ${ARCH} MATCHES "i386"))
+            # The pclmul code currently crashes on Mac in 32bit mode. Avoid for now.
+            check_c_source_compile_or_run(
+                "#include <immintrin.h>
+                int main(void)
+                {
+                    __m128i a = _mm_setzero_si128();
+                    __m128i b = _mm_setzero_si128();
+                    __m128i c = _mm_clmulepi64_si128(a, b, 0x10);
+                    (void)c;
+                    return 0;
+                }"
+                HAVE_PCLMULQDQ_INTRIN
+            )
+        else()
+            set(HAVE_PCLMULQDQ_INTRIN NO)
+        endif()
+        set(CMAKE_REQUIRED_FLAGS)
     endif()
-    check_c_source_compile_or_run(
-        "#include <immintrin.h>
-        int main(void) {
-            __m256i x = _mm256_set1_epi16(2);
-            const __m256i y = _mm256_set1_epi16(1);
-            x = _mm256_subs_epu16(x, y);
-            (void)x;
-            return 0;
-        }"
-        HAVE_AVX2_INTRIN
-    )
-    set(CMAKE_REQUIRED_FLAGS)
-
-    # FORCE_SSE2 option will only be shown if HAVE_SSE2_INTRIN is true
-    if("${ARCH}" MATCHES "i[3-6]86")
-        cmake_dependent_option(FORCE_SSE2 "Always assume CPU is SSE2 capable" OFF "HAVE_SSE2_INTRIN" OFF)
+    if(WITH_AVX2)
+        # Check whether compiler supports AVX2 intrinics
+        if(WITH_NATIVE_INSTRUCTIONS)
+            set(CMAKE_REQUIRED_FLAGS "${NATIVEFLAG}")
+        else()
+            set(CMAKE_REQUIRED_FLAGS "${AVX2FLAG}")
+        endif()
+        check_c_source_compile_or_run(
+            "#include <immintrin.h>
+            int main(void) {
+                __m256i x = _mm256_set1_epi16(2);
+                const __m256i y = _mm256_set1_epi16(1);
+                x = _mm256_subs_epu16(x, y);
+                (void)x;
+                return 0;
+            }"
+            HAVE_AVX2_INTRIN
+        )
+        set(CMAKE_REQUIRED_FLAGS)
```

Comment:
See above, cached variables must be explicitly cleared.

Reply:
Fixed.